charts/cloudcasa/v0.1.0/templates/cluster-register.yaml

```diff
@@ -0,0 +1,2978 @@
+                type: string
+            type: object
+        type: object
+    version: v1
+    versions:
+    - name: v1
+      served: true
+      storage: true
+- apiVersion: apiextensions.k8s.io/v1beta1
+  kind: CustomResourceDefinition
+  metadata:
+    annotations:
+      controller-gen.kubebuilder.io/version: v0.3.0
+    creationTimestamp: null
+    labels:
+      component: kubeagent-backup-helper
+    name: volumesnapshotlocations.velero.io
+  spec:
+    group: velero.io
+    names:
+      kind: VolumeSnapshotLocation
+      listKind: VolumeSnapshotLocationList
+      plural: volumesnapshotlocations
+      singular: volumesnapshotlocation
+    preserveUnknownFields: false
+    scope: Namespaced
+    validation:
+      openAPIV3Schema:
+        description: VolumeSnapshotLocation is a location where Velero stores volume snapshots.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: VolumeSnapshotLocationSpec defines the specification for a Velero VolumeSnapshotLocation.
+            properties:
+              config:
+                additionalProperties:
+                  type: string
+                description: Config is for provider-specific configuration fields.
+                type: object
+              provider:
+                description: Provider is the provider of the volume storage.
+                type: string
+            required:
+            - provider
+            type: object
+          status:
+            description: VolumeSnapshotLocationStatus describes the current status of a Velero VolumeSnapshotLocation.
+            properties:
+              phase:
+                description: VolumeSnapshotLocationPhase is the lifecyle phase of a Velero VolumeSnapshotLocation.
+                enum:
+                - Available
+                - Unavailable
+                type: string
+            type: object
+        type: object
+    version: v1
+    versions:
+    - name: v1
+      served: true
+      storage: true
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    creationTimestamp: null
+    labels:
+      component: kubeagent-backup-helper
+    name: cloudcasa-io
+    namespace: cloudcasa-io
```

Comment:
Rancher relies on Helm's built-in objects so that it can install the chart in its own namespace, or change the name, etc. So hardcoded namespaces and names like this could cause conflicts.
```yaml
  name: {{ .Release.Name }}
  namespace: {{ .Release.Namespace }}
```